dmoj/judge.py

```diff
@@ -459,69 +491,132 @@ def _grade_cases(self) -> Generator[Tuple[IPC, tuple], None, None]:
             if hasattr(binary, 'warning') and binary.warning is not None:
                 yield IPC.COMPILE_MESSAGE, (binary.warning,)
 
-        yield IPC.GRADING_BEGIN, (self.grader.run_pretests_only,)
+        class GradingAbort(Exception):
+            pass
+
+        def grade(case: AbstractTestCase) -> Result:
```

Comment:
```suggestion
        def grade_single(case: AbstractTestCase) -> Result:
```

To match the naming of the rest of the functions.